server/src-lib/Hasura/RQL/Types/RemoteSchema.hs

```diff
@@ -0,0 +1,84 @@
+{-# LANGUAGE DeriveLift        #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes       #-}
+{-# LANGUAGE TemplateHaskell   #-}
+{-# LANGUAGE TypeFamilies      #-}
+
+module Hasura.RQL.Types.RemoteSchema where
+
+import           Hasura.Prelude
+import           Language.Haskell.TH.Syntax (Lift)
+
+import qualified Data.Aeson                 as J
+import qualified Data.Aeson.Casing          as J
+import qualified Data.Aeson.TH              as J
+import qualified Data.HashMap.Strict        as Map
+import qualified Network.URI.Extended       as N
+
+import           Hasura.RQL.DDL.Headers     (HeaderConf (..))
+
+
+type UrlFromEnv = Text
+
+data RemoteSchemaDef
+  = RemoteSchemaDef
+  { _rsName             :: !Text
+  , _rsUrl              :: !(Either N.URI UrlFromEnv)
+  , _rsHeaders          :: ![HeaderConf]
+  , _rsFwdClientHeaders :: !Bool
+  } deriving (Show, Eq, Lift)
+
+type RemoteSchemaMap = Map.HashMap N.URI RemoteSchemaDef
```

Comment:
Should be `Map SchemaName RemoteSchemaDef`

Reply:
Have a `RemoteSchemaInfo` type which captures the final URI (either directly specified or read from an env var)